server/src/main/java/io/crate/execution/jobs/SharedShardContexts.java

```diff
@@ -90,16 +91,36 @@ public CompletableFuture<Void> maybeRefreshReaders(Metadata metadata,
         return CompletableFuture.allOf(refreshActions.toArray(CompletableFuture[]::new));
     }
 
+    /**
+     * Gets a context for the provided shardId and readerId.
+     * Creates the context if missing.
+     *
+     * This is intended to be called by {@link FetchTask} to create the contexts for shared by subsequent collect/fetch phases.
+     * Getting an existing context is an edge case for when there are multiple fetch tasks over the same query.
+     * (self-join involving virtual tables with limits and aggregations)
+     **/
     public SharedShardContext createContext(ShardId shardId, int readerId) throws IndexNotFoundException {
```

Comment:
I'd maybe rename this to `prepareContext` as it no longer always creates a new context, and it fits the intention (fetchTask is preparing the context for subsequent use by collect and fetch-operation)

Reply:
Done